clash-prelude/src/Clash/Sized/Internal/Index.hs

```diff
@@ -184,10 +191,28 @@ instance NFData (Index n) where
   -- NOINLINE is needed so that Clash doesn't trip on the "Index ~# Integer"
   -- coercion
 
-instance (KnownNat n, 1 <= n) => BitPack (Index n) where
-  type BitSize (Index n) = CLog 2 n
-  pack   = packXWith pack#
-  unpack = unpack#
+instance KnownNat n => BitPack (Index n) where
+  type BitSize (Index n) = BitSizeIndex n
+  pack   = case compareSNat (SNat @1) (SNat @n) of
+    SNatGT                   -> const 0
+    SNatLE | Dict <- fact @n -> packXWith pack#
+  unpack = case compareSNat (SNat @1) (SNat @n) of
+    SNatGT                   -> const undefined
```

Comment:
Could you replace this with an `errorX` call with a nice error message?

Reply:
I vote for `clashCompileError`

Reply:
Is it? We want Clash to always throw constructs like this away (because netlist says it is zero bits).

Reply:
Ahh yes, this case can happen and must be undefined. Yes I agree with `errorX`.

Reply:
Done.